RecoEgamma/EgammaTools/src/GBRForestTools.cc

```diff
@@ -27,13 +27,30 @@ namespace {
 
 };
 
-std::unique_ptr<const GBRForest> GBRForestTools::createGBRForest(const std::string &weightFile){
-  edm::FileInPath weightFileEdm(weightFile);
-  return GBRForestTools::createGBRForest(weightFileEdm);
+// Creates a pointer to new GBRForest corresponding to a TMVA weights file,
+// given a file path as string.  When the file is not found, also the same name
+// appended with ".gz" and ".gzip" is tried.  This allows the maintainters of
+// the data repositories to gzip weight files a posteriori without changing
+// anything else.
+std::unique_ptr<const GBRForest> GBRForestTools::createGBRForest(const std::string &weightFile,
+                                                                 std::vector<std::string> &varNames){
+  try {
```

Comment:
use of try-catch is strongly discouraged.
We should first consider a solution without it.

Reply:
Ok, thanks for telling me. I will think of a cleaner solution.

Reply:
which part throws? FileInPath or the createGBRForest call?

Reply:
It's FileInPath...